src/components/Listings/index.js

```diff
@@ -1,47 +1,119 @@
-import React, { useState, useMemo } from "react"
+import React, { useMemo, useReducer } from "react"
 import { Link } from "gatsby"
 import { connectToSpreadsheet } from "react-google-sheet-connector"
 import classnames from "classnames"
 import { useDebounce } from "use-debounce"
 
-import { NEEDS_SHEET_NAME } from "../../utils/listingUtils"
+import {
+  NEEDS_SHEET_NAME,
+  NEED_TYPES,
+  NEEDS_SHEET_COLUMN_INDICES,
+} from "../../utils/listingUtils"
 import useTextSearch from "../../utils/useTextSearch"
 import ListingResults from "../ListingResults"
 import cs from "./styles.module.css"
 
 const FULL_TEXT_SEARCH_KEYS = ["name"]
 
+/**
+ * A reducer which parses a row from the sheet and adds the needs data to the array of all needs
+ * TODO: implement all types and parse metadata for each card
+ * @param {*} result
+ * @param {*} row
+ * @param {*} index
+ */
+function parseRow(result, row, index) {
+  const rowNeeds = []
+  if (row[NEEDS_SHEET_COLUMN_INDICES.isFinancialNeed]) {
+    const parsedFinancialMetadata = {
+      // any data parsed out of the row that is needed by the financial card
+      frequency: row[NEEDS_SHEET_COLUMN_INDICES.financial_needFrequency],
+      timing: row[NEEDS_SHEET_COLUMN_INDICES.financial_needTiming],
+      minfundingNeeded:
+        row[NEEDS_SHEET_COLUMN_INDICES.financial_minFundingNeed],
+      maxFundingNeeded:
+        row[NEEDS_SHEET_COLUMN_INDICES.financial_maxFundingNeed],
+      fundingMethod: row[NEEDS_SHEET_COLUMN_INDICES.financial_fundingMethod],
```

Comment:
This is the piece I was referencing would add a third piece of changing when a question is added/changed in the spreadsheet.

See: https://github.com/mmoutenot/mamas-needs-offers/pull/11#discussion_r403744545

Reply:
I think we need to assume this is going to be brittle to additions / changes regardless of how we do it. I like your approach of using indexes because it is stable as long as the order of questions / answers doesn't change. This PR solely moves the parsing logic to a centralized place rather than having it at the individual card level. We need to do the parsing centralized because we have to do filtering and searching on the parsed results.